### PR TITLE
Fix error handling when trying to connect to server

### PR DESCRIPTION
**Pull request by aelred**

### CODE REVIEW

**Comment by aelred** [comment]
@fppt I am now semi-confident this will fix `stable` 👍

File: grakn-client/src/main/java/ai/grakn/client/Client.java [hunk 1]
```diff
@@ -121,38 +122,58 @@ private static EngineStatus checkServerRunning() throws IOException {
      * @return true if Grakn Engine running, false otherwise
      */
     public static boolean serverIsRunning(SimpleURI uri) {
+        URL url;
         try {
-            URL url = UriBuilder.fromUri(uri.toURI()).path(REST.WebPath.KB).build().toURL();
+            url = UriBuilder.fromUri(uri.toURI()).path(REST.WebPath.KB).build().toURL();
+        } catch (MalformedURLException e) {
+            assert false : "This will never throw because we're appending a known path to a valid URI";
```
Comment (fppt) [inline grakn-client/src/main/java/ai/grakn/client/Client.java]: What magic is this @aelred ?

File: grakn-test/test-integration/src/test/java/ai/grakn/test/rule/DistributionContext.java [hunk 2]
```diff
@@ -59,15 +61,17 @@
     public static final Logger LOG = LoggerFactory.getLogger(DistributionContext.class);
 
     private static final FilenameFilter jarFiles = (dir, name) -> name.toLowerCase().endsWith(".jar");
-    private static final String ZIP = "grakn-dist-" + GraknVersion.VERSION + ".zip";
-    private static final String CURRENT_DIRECTORY = GraknSystemProperty.PROJECT_RELATIVE_DIR.value();
-    private static final String TARGET_DIRECTORY = CURRENT_DIRECTORY + "/grakn-dist/target/";
-    private static final String DIST_DIRECTORY = TARGET_DIRECTORY + "grakn-dist-" + GraknVersion.VERSION;
+    private static final Path ZIP = Paths.get("grakn-dist-" + GraknVersion.VERSION + ".zip");
```
Comment (fppt) [inline grakn-test/test-integration/src/test/java/ai/grakn/test/rule/DistributionContext.java]: yay!

File: grakn-test/test-integration/src/test/java/ai/grakn/test/rule/DistributionContext.java [hunk 3]
```diff
@@ -88,10 +92,7 @@ public SimpleURI uri(){
 
     @Override
     protected List<TestRule> testRules() {
-        return ImmutableList.of(
-                SessionContext.create(),
-                InMemoryRedisContext.create(redisPort)
-        );
+        return ImmutableList.of(session, redis);
```
Comment (fppt) [inline grakn-test/test-integration/src/test/java/ai/grakn/test/rule/DistributionContext.java]: This is kind of funny when you really think about it :D

File: grakn-client/src/main/java/ai/grakn/client/Client.java (hunk 1 above)
Comment (aelred) [reply grakn-client/src/main/java/ai/grakn/client/Client.java]: This is just an assertion that always fails. Basically we should never hit this code. OH I JUST REMEMBERED WE HAVE `CommonUtil.unreachableStatement()` FOR THIS

File: grakn-test/test-integration/src/test/java/ai/grakn/test/rule/DistributionContext.java (hunk 3 above)
Comment (aelred) [reply grakn-test/test-integration/src/test/java/ai/grakn/test/rule/DistributionContext.java]: Haha 😢

**Pull request merged**